src/TransferProxy.sol

```diff
@@ -72,13 +72,48 @@ contract TransferProxy is Auth, ITransferProxy {
     ERC20(token).safeTransferFrom(from, to, amount);
   }
 
+  function trySafeTransferFrom(
+    address token,
+    address from,
+    address to,
+    uint256 amount
+  ) internal returns (bool success) {
+    assembly {
+      // Get a pointer to some free memory.
+      let freeMemoryPointer := mload(0x40)
+
+      // Write the abi-encoded calldata into memory, beginning with the function selector.
+      mstore(
+        freeMemoryPointer,
+        0x23b872dd00000000000000000000000000000000000000000000000000000000
+      )
+      mstore(add(freeMemoryPointer, 4), from) // Append the "from" argument.
+      mstore(add(freeMemoryPointer, 36), to) // Append the "to" argument.
+      mstore(add(freeMemoryPointer, 68), amount) // Append the "amount" argument.
+
+      success := and(
+        // Set success to whether the call reverted, if not we check it either
+        // returned exactly 1 (can't just be non-zero data), or had no return data.
+        or(
+          and(eq(mload(0), 1), gt(returndatasize(), 31)),
+          iszero(returndatasize())
+        ),
+        // We use 100 because the length of our calldata totals up like so: 4 + 32 * 3.
+        // We use 0 and 32 to copy up to 32 bytes of return data into the scratch space.
+        // Counterintuitively, this call must be positioned second to the or() call in the
+        // surrounding and() call or else returndatasize() will be zero during the computation.
+        call(gas(), token, 0, freeMemoryPointer, 100, 0, 32)
```

Comment:
just a note that after this point `freeMemoryPointer` does not point to a clean memory region.